packages/keyValueCache/src/InMemoryLRUCache.ts

```diff
@@ -0,0 +1,58 @@
+import LRUCache from "lru-cache";
+import type { KeyValueCache, KeyValueCacheSetOptions } from "./KeyValueCache";
+
+// LRUCache wrapper to implement the KeyValueCache interface.
+export class InMemoryLRUCache<T = string> implements KeyValueCache<T> {
+  private cache: LRUCache<string, T>;
+
+  constructor(lruCacheOpts?: LRUCache.Options<string, T>) {
+    this.cache = new LRUCache({
+      sizeCalculation,
+      // Create ~about~ a 30MiB cache by default. Configurable by providing
+      // `lruCacheOpts`.
+      maxSize: Math.pow(2, 20) * 30,
+      ...lruCacheOpts,
+    });
+  }
+
+  async set(key: string, value: T, options?: KeyValueCacheSetOptions) {
+    if (options?.ttl) {
+      this.cache.set(key, value, { ttl: options.ttl * 1000 });
+    } else {
+      this.cache.set(key, value);
+    }
+  }
+
+  async get(key: string) {
+    return this.cache.get(key);
+  }
+
+  async delete(key: string) {
+    return this.cache.delete(key);
+  }
+
+  clear() {
+    this.cache.clear();
+  }
+
+  keys() {
+    const keys = [];
+    for (const key of this.cache.keys()) {
```

Comment:
I think the fact that you're not just doing `return this.cache.keys()` deserves a comment.

(And can this just use `[... this.cache.keys()]`?)